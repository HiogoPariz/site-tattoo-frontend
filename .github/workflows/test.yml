name: Test workflow

on:
  pull_request:
    branches: [staging]
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Check out
        uses: actions/checkout@v2

      - name: Set up Node 14
        uses: actions/setup-node@v3
        with:
          node-version: '14'
          cache: 'yarn'
          cache-dependency-path: yarn.lock

      - name: Install dependencies
        run: yarn install --frozen-lockfile

      - name: Test with jest
        run: yarn test --ci --maxWorkers=100%
