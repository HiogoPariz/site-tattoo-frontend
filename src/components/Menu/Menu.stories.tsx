import React from 'react'
import { Meta, Story } from '@storybook/react'

import { Menu, MenuProps } from './Menu'

const meta: Meta = {
  title: 'Components/Menu',
  component: Menu
}

export default meta
const Template: Story<MenuProps> = args => <Menu {...args} />

export const Default = Template.bind({})
