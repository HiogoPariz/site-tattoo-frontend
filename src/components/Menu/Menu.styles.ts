import styled from 'styled-components'

export const MenuContainer = styled.header`
  display: flex;
  width: 100%;
  grid-area: menu;
  justify-content: center;
  padding: ${({ theme }) => theme.spacings.normal};
`

export const MenuForm = styled.form`
  display: flex;
  border-radius: ${({ theme }) => theme.borderRadius.card};
  padding: ${({ theme }) => theme.spacings.normal};
`

export const Input = styled.input``

export const MainMenu = styled.div`
  border-radius: ${({ theme }) => theme.borderRadius.card};
  height: 2rem;
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spacings.big};
`
