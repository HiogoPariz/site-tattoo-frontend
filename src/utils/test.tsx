/* eslint-disable import/export */
import React from 'react'
import { render } from '@testing-library/react'
import theme from '@/styles/theme'
import { ThemeProvider } from 'styled-components'

// Add in any providers here if necessary:
// (ReduxProvider, ThemeProvider, etc)
const Providers: React.FC = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
