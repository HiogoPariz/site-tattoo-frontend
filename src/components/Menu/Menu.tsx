import React from 'react'
import { NavLink } from '@/types'
import { Input, MainMenu, MenuContainer, MenuForm } from './Menu.styles'
import Link from 'next/link'
import FocusContainer from '@/styles/containers/focus'

export type MenuProps = {
  links: NavLink[]
}

export const Menu: React.FC<MenuProps> = () => {
  return (
    <MenuContainer>
      <MainMenu>
        <Link href="/">Logo</Link>
        <MenuForm>
          <FocusContainer>
            teste
            <Input />
          </FocusContainer>
          <FocusContainer>
            teste2
            <Input />
          </FocusContainer>
        </MenuForm>
      </MainMenu>
    </MenuContainer>
  )
}
