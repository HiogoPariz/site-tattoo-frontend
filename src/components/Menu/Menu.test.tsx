import React from 'react'
import { render } from '@/utils/test'
import { Menu, MenuProps } from './Menu'

function setup(props?: Partial<MenuProps>) {
  const mock: MenuProps = {
    links: [
      { label: 'Home', to: '/' },
      { label: 'Teste', to: '/teste' }
    ],
    ...props
  }
  return { mock, ...render(<Menu {...mock} />) }
}

describe('<Menu />', () => {
  it('should render', () => {
    const { container } = setup()
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toBeInTheDocument()
  })
})
