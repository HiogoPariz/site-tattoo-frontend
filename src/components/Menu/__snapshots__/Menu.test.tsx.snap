// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  grid-area: menu;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 1rem;
  padding: 1.5rem;
}

.c1 {
  border-radius: 1rem;
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
}

.c3 {
  border: 0.1rem solid #A75E55;
  -webkit-transition: border 200ms;
  transition: border 200ms;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3:focus,
.c3:active {
  border: 0.1rem solid #6D5063;
}

<header
  class="c0"
>
  <div
    class="c1"
  >
    <a
      href="/"
    >
      Logo
    </a>
    <form
      class="c2"
    >
      <label
        class="c3"
      >
        teste
        <input
          class="Menustyles__Input-sc-1pqmg6w-2"
        />
      </label>
      <label
        class="c3"
      >
        teste2
        <input
          class="Menustyles__Input-sc-1pqmg6w-2"
        />
      </label>
    </form>
  </div>
</header>
`;
